src/Gaffer/ScriptNode.cpp

```diff
@@ -534,48 +534,46 @@ void ScriptNode::paste( Node *parent )
 	}
 }
 
+namespace
+{
+
 void handleReconnectionOfOutputPlug( const ScriptNode *script, const DependencyNode *dependencyNode, const Plug *outputPlug )
 {
-	// for plugs that hold output plugs we need to recurse
-	if(outputPlug->typeId() == Plug::staticTypeId())
+	const Plug *inPlug = dependencyNode->correspondingInput( outputPlug );
+	if ( !inPlug )
 	{
 		for( OutputPlugIterator it( outputPlug ); !it.done(); ++it )
 		{
 			handleReconnectionOfOutputPlug( script, dependencyNode, it->get() );
 		}
-		return;
 	}
-
-	const Plug *inPlug = dependencyNode->correspondingInput( outputPlug );
-	if ( !inPlug )
+	else
 	{
-		return;
-	}
-
-	const Plug *srcPlug = inPlug->getInput<Plug>();
-	if ( !srcPlug )
-	{
-		return;
-	}
-
-	// record this plug's current outputs, and reconnect them. This is a copy of outputPlug->outputs() rather
-	// than a reference, as reconnection can modify outputPlug->outputs()...
-	Plug::OutputContainer outputs = outputPlug->outputs();
-	for ( Plug::OutputContainer::const_iterator oIt = outputs.begin(); oIt != outputs.end(); )
-	{
-		Plug *dstPlug = *oIt;
-		if ( dstPlug && dstPlug->acceptsInput( srcPlug ) && script->isAncestorOf( dstPlug ) )
+		const Plug *srcPlug = inPlug->getInput<Plug>();
+		if ( !srcPlug )
 		{
-			oIt++;
-			dstPlug->setInput( const_cast<Gaffer::Plug*>(srcPlug) );
+			return;
```

Comment:
This `return` doesn't feel right to me - if we don't have a connection at this level, we may still have children for which `correspondingInput()` returns something which has a connection. As I said in my previous comment, I think what we need to do is move the recurse-to-children code to the end of the function, and only return early (instead of recursing) when we successfully make a reconnection for the current plug. It'd be good to update the test case to show this working - probably with "out", "out.a" and "out.b" with corresponding inputs of "in", "in.a" and "in.b", testing that everything works when the existing connections are at the "in/out" level and also with a single connection at the "in.a" level.